product_listprice_list_view/models/product_template.py

```diff
@@ -87,6 +91,29 @@ def _get_local_location_name(self, prod_ids):
         )
         return quant.location_id.name
 
+    def _get_overseas_note(self):
+        for pt in self:
+            # Ids of all  products variants of that template,
+            # which is actually always 1 because we do not have product variants
+            prod_ids = [p.id for p in pt.product_variant_ids]
+            # Are among those ids overseas_stock?
+            if pt.overseas_stock == 'Yes':
+                ss_obj = self.env['supplier.stock']
+                # Supplier Stock among those ids
+                ss_recs = ss_obj.search(
+                    [('product_id', '=', prod_ids)]
+                )
+                lowest_cost = 0.0
+                lowest_cost_ss_rec = False
+                for ss_rec in ss_recs:
+                    if not lowest_cost or ss_rec.price_unit_base < lowest_cost:
+                        lowest_cost = ss_rec.price_unit_base
+                        lowest_cost_ss_rec = ss_rec
+                if lowest_cost_ss_rec:
+                    pt.partners_note = lowest_cost_ss_rec.partners_note
```

Comment:
Not knowing the full context, wouldn't it be better to extend `_get_stock_location` and `_get_overseas_location_name` methods instead of adding this new method?

Reply:
Please review the changes.
Thanks.